Terminal.Gui/ConsoleDrivers/NetDriver.cs

```diff
@@ -300,808 +304,224 @@ void GetConsoleInputType (ConsoleKeyInfo consoleKeyInfo)
 			inputResultQueue.Enqueue (inputResult);
 		}
 
-		void SplitCSI (ConsoleKeyInfo [] cki, ref InputResult inputResult, ref ConsoleKeyInfo newConsoleKeyInfo, ref ConsoleKey key, ref MouseEvent mouseEvent, ref ConsoleModifiers mod)
-		{
-			ConsoleKeyInfo [] splitedCki = new ConsoleKeyInfo [] { };
-			int length = 0;
-			var kChar = GetKeyCharArray (cki);
-			var nCSI = GetNumberOfCSI (kChar);
-			int curCSI = 0;
-			char previousKChar = '\0';
-			if (nCSI > 1) {
-				for (int i = 0; i < cki.Length; i++) {
-					var ck = cki [i];
-					if (NumberOfCSI > 0 && nCSI - curCSI > NumberOfCSI) {
-						if (i + 1 < cki.Length && cki [i + 1].KeyChar == '\x1b' && previousKChar != '\0') {
-							curCSI++;
-							previousKChar = '\0';
-						} else {
-							previousKChar = ck.KeyChar;
-						}
-						continue;
-					}
-					if (ck.KeyChar == '\x1b') {
-						if (ck.KeyChar == 'R') {
-							ResizeArray (ck);
-						}
-						if (splitedCki.Length > 1) {
-							DecodeCSI (ref inputResult, ref newConsoleKeyInfo, ref key, ref mouseEvent, splitedCki, ref mod);
-						}
-						splitedCki = new ConsoleKeyInfo [] { };
-						length = 0;
-					}
-					ResizeArray (ck);
-					if (i == cki.Length - 1 && splitedCki.Length > 0) {
-						DecodeCSI (ref inputResult, ref newConsoleKeyInfo, ref key, ref mouseEvent, splitedCki, ref mod);
-					}
-				}
-			} else {
-				DecodeCSI (ref inputResult, ref newConsoleKeyInfo, ref key, ref mouseEvent, cki, ref mod);
-			}
-
-			void ResizeArray (ConsoleKeyInfo ck)
-			{
-				length++;
-				Array.Resize (ref splitedCki, length);
-				splitedCki [length - 1] = ck;
-			}
-		}
-
-		char [] GetKeyCharArray (ConsoleKeyInfo [] cki)
+		void DecodeEscSeq (ref ConsoleKeyInfo newConsoleKeyInfo, ref ConsoleKey key, ConsoleKeyInfo [] cki, ref ConsoleModifiers mod)
 		{
-			char [] kChar = new char [] { };
-			var length = 0;
-			foreach (var kc in cki) {
-				length++;
-				Array.Resize (ref kChar, length);
-				kChar [length - 1] = kc.KeyChar;
-			}
-
-			return kChar;
-		}
+			string c1Control, code, terminating;
+			string [] values;
+			bool isKeyMouse, isReq;
```

Comment:
What does `isKeyMouse` mean? Can you choose a name that's more obvious, or add a comment?

Reply:
Ah. Got it. THanks!